lib/yarn-2.sh

```diff
@@ -28,7 +28,15 @@ has_yarn_cache() {
 has_yarn_workspace_plugin_installed() {
   local build_dir="$1"
   local plugin_paths=$(get_yarn_plugin_paths "$build_dir")
-  [[ ! -n $plugin_paths ]] && echo "$plugin_paths" | grep plugin-workspace-tools
+  local path
+
+  if [[ -n $plugin_paths ]]; then
+    path=$(echo "$plugin_paths" | grep plugin-workspace-tools)
+  else
+    path=""
+  fi
+
+  [[ ! "$path" == "" ]]
```

Comment:
I'd recommend instead querying Yarn directly to see if the plugin is installed, to guard against potential changes in config format or location. Something like:

```
yarn plugin runtime | grep plugin-workspace-tools
```